ext/openssl/ossl_pkey_ec.c

```diff
@@ -1364,6 +1374,120 @@ static VALUE ossl_ec_point_set_to_infinity(VALUE self)
     return self;
 }
 
+#ifdef HAVE_EC_POINT_GET_AFFINE_COORDINATES
+
+/*
+ * call-seq:
+ *   point.affine_coords => [ xBN, yBN ]
+ */
+static VALUE
+ossl_ec_point_get_affine_coords(VALUE self)
+{
+    VALUE x, y, result;
+    EC_POINT *point;
+    BIGNUM *xBN, *yBN;
+    const EC_GROUP *group;
+
+    GetECPoint(self, point);
+    GetECPointGroup(self, group);
+
+    if (point == NULL || group == NULL) {
+        rb_raise(eEC_POINT, "unable to get point and group");
+    }
+
+    x = ossl_bn_new(NULL);
+    y = ossl_bn_new(NULL);
+
+    xBN = GetBNPtr(x);
+    yBN = GetBNPtr(y);
+
+    if (!EC_POINT_get_affine_coordinates(group, point, xBN, yBN, NULL)) {
+        ossl_raise(eEC_POINT, "EC_POINT_get_affine_coordinates");
+    }
+
+    result = rb_ary_new2(2);
+    rb_ary_push(result, x);
+    rb_ary_push(result, y);
+
+    return result;
+}
+#endif
+
+#ifdef HAVE_EC_POINT_SET_AFFINE_COORDINATES
+/*
+ * call-seq:
+ *   point.affine_coords = [ xBN, yBN ]
+ */
+static VALUE
+ossl_ec_point_set_affine_coords(VALUE self, VALUE coords)
+{
+    VALUE x, y;
+    EC_POINT *point;
+    BIGNUM *xBN, *yBN;
+    const EC_GROUP *group;
+
+    GetECPoint(self, point);
+    GetECPointGroup(self, group);
+
+    if (point == NULL || group == NULL) {
+        rb_raise(eEC_POINT, "unable to get point and group");
+    }
+
+    if (TYPE(coords) != T_ARRAY || RARRAY_LEN(coords) != 2) {
+        rb_raise(eEC_POINT, "argument must be an array of [ x, y ]");
+    }
+
+    x = ossl_try_convert_to_bn(RARRAY_AREF(coords, 0));
+    y = ossl_try_convert_to_bn(RARRAY_AREF(coords, 1));
+
+    xBN = GetBNPtr(x);
+    yBN = GetBNPtr(y);
```

Comment:
`GetBNPtr()` uses `try_convert_to_bn()` internally. So, simply `x = RARRAY_AREF(coords, 0);` followed by `xBN = GetBNPtr(x)` should be enough.

The value from the array still needs to be stored into a separate VALUE variable (`x`, `y`) once, though.